Docs/Book/GettingStartedInC++.md

```diff
@@ -0,0 +1,1420 @@
+#include <GraphMol/ForceFieldHelpers/MMFF/MMFF.h>
+.
+.
+RDKit::MMFF::MMFFOptimizeMolecule( *mol2 , 1000 , "MMFF94s" );
+```
+Please note that the MMFF atom typing code uses its own aromaticity model,
+so the aromaticity flags of the molecule will be modified after calling
+MMFF-related methods.
+Note the calls to `RDKit::MolOps::addHs()` in the examples above. By
+default RDKit molecules do not have H atoms explicitly present in the
+graph, but they are important for getting realistic geometries, so
+they generally should be added.  They can always be removed afterwards
+if necessary with a call to `RDKit::MolOps::removeHs()`
+
+With the RDKit, multiple conformers can also be generated using the two
+different embedding methods. In both cases this is simply a matter of
+running the distance geometry calculation multiple times from
+different random start points. The 2nd parameter to
+`EmbedMultipleConfs` allows the user to
+set the number of conformers that should be generated.  Otherwise the
+procedures are similar to before
+[(example11.cpp)](./C++Examples/example11.cpp):
+```c++
+RDKit::INT_VECT mol1_cids = RDKit::DGeomHelpers::EmbedMultipleConfs( *mol1 , 10 );
+std::cout << "Number of conformations : " << mol1_cids.size() << std::endl;
+
+RDKit::INT_VECT mol2_cids;
+RDKit::DGeomHelpers::EmbedMultipleConfs( *mol2 , mol2_cids , 20 , 1 , 30 , 1234 ,
+                                         true , false , 2.0 , true , 1 , -1.0 , 
+                                         static_cast<const std::map<int,RDGeom::Point3D> *> ( 0 ) ,
+                                         1e-3 , false , true , true , true );
+std::cout << "Number of conformations : " << mol2_cids.size()
+          << std::endl;
+```
+The conformer ids are returned in `mol1_cids` and `mol2_cids` and
+there are two overloaded functions with different ways of supplying
+the information.  As before, the CSD-based method is invoked by the
+last two `true` parameters, and in the example above the default
+number of conformations to be produced has been changed from 10 to 20.
+The conformers so generated can be aligned
+to each other and the RMS values calculated
+[(example11.cpp)](./C++Examples/example11.cpp):
+```c++
+#include <GraphMol/MolAlign/AlignMolecules.h>
+.
+.
+std::vector<double> rms_list;
+std::vector<unsigned int> m2cids( mol2_cids.begin() , mol2_cids.end() );
+RDKit::MolAlign::alignMolConformers( *mol2 ,
+                                     static_cast<const std::vector<unsigned int> *>( 0 ) ,
+                                     &m2cids ,
+                                     static_cast<const RDNumeric::DoubleVector *>( 0 ) ,
+                                     false , 50 , &rms_list );
+```
+The RMS values for the overlays will be fed into rms_list on return.
+Note the somewhat inconvenient issue that `EmbedMultipleConfs` returns
+a vector of `ints` for the conformer ids, but `alignMolConformers`
+requires a vector of `unsigned ints`. The first vector of `unsigned
```

Comment:
inconvenient perhaps, but there is a reason for it. EmbedMultipleConfs() needs to be able to return -1 to indicate failure.